msprime/simulations.py

```diff
@@ -1471,6 +1489,198 @@ def print_history(self, output=sys.stdout):
             self._print_populations(epoch, output)
             print(file=output)
 
+    def population_size_trajectory(self, steps):
+        """
+        This function returns an array of per-population effective population sizes,
+        as defined by the demographic model. These are just the `population_size`
+        parameters of the model, modified by any population growth rates.
+        The sizes are computed at a regular grid of time steps, between time 0 and `end`
+        units of time ago.
+
+        :param int end: Time (ago) at which to stop trajectory inference.
+
+        :param int num_steps: determines at how many time points the
+            coalescence rate is computed.
+
+        :return: Returns a tuple where the first value
+            is an array giving the times at which population sizes have been computed,
+            the second is a num_steps-by-num_pops ndarray containing the effective
+            population sizes for each subpopulation individually,
+            at the corresponding step
+
+        :rtype: :class:'tuple'
+        """
+        num_pops = self.num_populations
+        N_t = np.zeros([len(steps), num_pops])
+        for j, t in enumerate(steps):
+            N, _ = self.pop_size_and_migration_at_t(t)
+            N_t[j] = N
+        return N_t
+
+    def coalescence_rate_trajectory(self, steps, num_samples, min_pop_size=1,
+                                    double_step_validatation=True):
+        """
+        This function will calculate the mean coalescence rates between the
```

Comment:
"This function will calculate the mean coalescence rates and proportions of uncoalesced lineages ..."